### PR TITLE
[nemo-systemsettings] Add preferred language support. Fixes JB#54978

### PR DESCRIPTION
**Pull request by DieTime**

### CODE REVIEW

**Comment by pvuorela** [comment]
Does this really need new API or even configuration? E.g. perhaps could just read system locale for the initial index value?

**Comment by dseight** [comment]
`/etc/locale.conf` represents a current system-wide locale, which [could be changed](https://github.com/sailfishos/nemo-qml-plugin-systemsettings/blob/fc4cd71512c5b5232cf074bdfa593abf424d1379/setlocale/main.cpp#L134-L143) by the device owner. But the "preferred" locale must stay unchanged under any circumstances. It's kinda "vendor preference" or "customer preference".

File: src/languagemodel.h [hunk 1]
```diff
@@ -57,6 +57,7 @@ class SYSTEMSETTINGS_EXPORT LanguageModel: public QAbstractListModel
 {
     Q_OBJECT
     Q_PROPERTY(int currentIndex READ currentIndex NOTIFY currentIndexChanged)
+    Q_PROPERTY(int preferredIndex READ preferredIndex)
```
Comment (pvuorela) [inline src/languagemodel.h:60]: Properties should be either having a change signal or marked as CONSTANT

File: src/localeconfig.cpp [hunk 2]
```diff
@@ -49,3 +49,9 @@ QString systemLocaleConfigPath()
     // System-wide locale config
     return QString("/etc/locale.conf");
 }
+
+QString preferredLocaleConfigPath()
+{
+    // Prefer-wide locale config
```
[the remark this answers is not in the document]
Comment (DieTime) [reply src/localeconfig.cpp]: Oops. My mistake :D

**Comment by DieTime** [comment]
> Thanks. Looks now otherwise good but the commit message is missing the bug reference. Could you still add that there.

Done

**Pull request merged**